File: chapter7/typescript/SushiKitchen.ts
import { Kitchen } from "./Kitchen";

/**
 * Sushi Kitchen will prepare and make Sushi per customer order menu
 *
 * @version 1.0.0
 * @Author Bridget
 * @license MIT
 */
 export  class SushiKitchen implements Kitchen {
    /**
     * make sushi food based on customer order items
     *
     * @param items
     */
    public  makeFood(items: String[]): void{
        if (items && items.length>0) {
            console.log("make Sushi:[");
            for (let item of items) {
                console.log(" {" + item + "} ");
            }
            console.log("]");
        }

    }
}

File: chapter7/java/designpattern/src/main/java/com/designpattern/SushiKitchen.java
package com.designpattern;

import java.util.List;

/**
 * Sushi Kitchen will prepare and make Sushi per customer order menu
 *
 * @version 1.0.0
 * @Author Bridget
 * @license MIT
 */
public class SushiKitchen implements Kitchen {
    /**
     * make sushi food based on customer order items
     *
     * @param items
     */
    @Override
    public void makeFood(List<String> items) {
        if (!items.isEmpty()) {
            System.out.println("make Sushi:[");
            for (String item : items) {
                System.out.println(" {" + item + "} ");
            }
            System.out.println("]");
        }
    }
}

